src/validation.cpp

```diff
@@ -3016,21 +3016,13 @@ static bool FindBlockPos(CDiskBlockPos &pos, unsigned int nAddSize, unsigned int
         vinfoBlockFile[nFile].nSize += nAddSize;
 
     if (!fKnown) {
-        unsigned int nOldChunks = (pos.nPos + BLOCKFILE_CHUNK_SIZE - 1) / BLOCKFILE_CHUNK_SIZE;
-        unsigned int nNewChunks = (vinfoBlockFile[nFile].nSize + BLOCKFILE_CHUNK_SIZE - 1) / BLOCKFILE_CHUNK_SIZE;
```

Comment:
In commit "validation: Refactor block file pre-allocation into FlatFileSeq." (72d693209018fc0b94d7875574686283936751e6)

Note (just for reference because I found this function confusing): In the new code this line is replaced with:

```c++
unsigned int n_new_chunks = (pos.nPos + add_size + m_chunk_size - 1) / m_chunk_size;
```

which is equivalent because `vinfoBlockFile[nFile].nSize` == `pos.nPos + add_size` in the `!fKnown` case due to:

```c++
pos.nPos = vinfoBlockFile[nFile].nSize;
...
vinfoBlockFile[nFile].nSize += nAddSize;		￼
```

above

Reply:
Probably the right move would be to add some comments in the new code, to reduce confusion to future readers?

Reply:
I'm happy to add comments anywhere you think it would make the new code more clear, but I don't want to add comments because the old code was unnecessarily confusing and this is a slight change for anyone who happened to be familiar with it. I personally find the new Allocate function clear enough as is (including the `n_new_chunks` calculation), but of course I wrote it...

A PR comment like the one above seems like a fine way to address this sort of thing IMO.